sortinghat/core/importer/backends/openinfra.py

```diff
@@ -0,0 +1,188 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Jose Javier Merchante <jjmerchante@bitergia.com>
+#
+
+import datetime
+import logging
+
+import dateutil.tz
+import requests
+
+from grimoirelab_toolkit.datetime import str_to_datetime
+from grimoirelab_toolkit.uris import urijoin
+from sortinghat.core.errors import LoadError
+from sortinghat.core.importer.backend import IdentitiesImporter
+from sortinghat.core.importer.models import (Individual,
+                                             Identity,
+                                             Enrollment,
+                                             Organization,
+                                             Profile)
+
+logger = logging.getLogger(__name__)
+
+
+class OpenInfraIDImporter(IdentitiesImporter):
+
+    NAME = 'OpenInfraID'
+
+    def __init__(self, ctx, url, from_date=None):
+        super().__init__(ctx, url)
+        if isinstance(from_date, str):
+            from_date = str_to_datetime(from_date)
+        self.from_date = from_date
+
+    def get_individuals(self):
+        parser = OpenInfraIDParser(self.url)
+        return parser.individuals(self.from_date)
+
+
+class OpenInfraIDParser:
+    """Parse identities and organizations from OpenInfraID API.
+
+    The OpenInfraID data is from and API that provides the
+    identities and organizations of all the individuals.
+
+    The individuals are stored in an object named 'individuals'.
+    The keys of this object are the UUID of the individuals.
+    Each individual object stores a list of identities and
+    enrollments.
+
+    :param url: OpenInfraID API URL
+    :param last_update: datetime of the last update in UTC format
+
+    :raises InvalidFormatError: raised when the format of the stream is
+        not valid.
+    """
+
+    # API path
+    MEMBERS = '/api/public/v1/members'
+
+    # Resource parameters
+    PPER_PAGE = 'per_page'
+    PPAGE = 'page'
+    PSORT = 'sort'
+    PFILTER = 'filter'
+
+    def __init__(self, url):
+        self.url = url
+        self.source = 'openinfra'
+
+    def individuals(self, from_date=None):
+        """Fetch individuals from the OpenInfraID API
+
+        This method returns an iterator of individuals. Each
+        individual can contain an OpenInfraID identity which
+        username is the ID of the member, a GitHub identity,
+        and the enrollment of that identity.
+
+        The OpenInfraID members that don't contain any information
+        like name, GitHub identity or enrollments are skipped.
+
+        :param from_date: obtain members updated since this date
+
+        :returns: a generator of individuals
+        """
+
+        for member in self.fetch_members(from_date):
+            uuid = member.get('id')
+            name = f"{member.get('first_name', '')} {member.get('last_name', '')}".strip()
+            gender = member.get('gender')
```

Comment:
I think we shouldn't import the gender data even if it's there. I don't want people to get in trouble importing this kind of data. People using this importer might not known this data is there.